### DISCUSSION TITLE
Replace cached data by key (SSE)

### DISCUSSION THREAD

**Discussion by lacorde** [nuxt/nuxt#17578] [Questions]
I've been struggling to use Server-Side Events to update data in real time.

Currently, I use **useAsyncData**, wrapping an **axios instance**, see below :

```
const { data: order } = await useAsyncData(
  "/api/orders/" + id,
  () => $axios.get("/api/orders/" + id)
);
```
When an **order** entity is updated in our database, the server sends me a notification, with the updated JSON object as payload.

Now let's say the `order` with id `1` is updated.

How can I replace the data cached with the key `/api/orders/1` ?

Also, what is Nuxt's philosophy regarding SSE ?
Thank you !

PS: I know about `refresh` and `refreshNuxtData`, but the point here is not having to make another call !

**Comment by lewebsimple** [comment]
Until https://github.com/unjs/h3/issues/85 is resolved I believe SSE is out of the question for Nuxt3.
Although this recent [experimental feature ](https://v3.nuxtjs.org/guide/features/server-routes/#sending-streams-experimental) based on defineEventHandler might hold a clue as to how to achieve this.

**Reply by lacorde** [reply]
Thank you for your answer, I will keep an eye on this.

If it can help anyone, I ended up calling `refreshNuxtData` on whichever keys my SSE stream noticed there was an update for. Kinda overkill, but does the job.

Also [this SSE interface plugin](https://github.com/tserkov/vue-sse) works great on Nuxt3/Vue3 (not that you would need any plugins for that, but still).
